### PR TITLE
Add endpoint to set default campaign code

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Add a new endpoint to be able to attatch a campaign code to the current or new shop session.

## Justify why they are needed

This will be used in marketting.

We discussed that limiting it to a given endpoint will help us to maintain it rather than supporting a query parameter anywhere on the site. This was okeyed by Simon and Sonny.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1496** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1496" target="_blank">Graphite</a> 
    * **PR #1499** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1499" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1499?utm_source=stack-comment).

File: apps/store/src/pages/api/campaign/[code].ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ */
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { query, url } = req
+  let fallbackRedirect = PageLink.home()
+
+  const nextURL = new URL(ORIGIN_URL)
+  const nextQueryParam = query['next']
+  if (typeof nextQueryParam !== 'string') {
+    console.error('Missing next query parameter: ', url)
+    return res.redirect(302, fallbackRedirect)
```
Comment (alebedev) [inline apps/store/src/pages/api/campaign/[code].ts:25]: We should probably use HTTP 307 for temporary redirects, same as NextJs

**Review by alebedev — APPROVED** [review]
Looks good

**Comment by robinandeer** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: oaw3UqsiPM7uMcugvuzG.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1499)

**Pull request merged**